### ISSUE TITLE
Training a network with fit generator

### ISSUE THREAD

**Issue by cisi17** [keras-team/keras#9858]
Hello,

I am training a neural network with fit generator because of the large dataset I am using. This dataset consist of 56000 samples and has got 60 classes. The line that I use to train is the following one:

history=model.fit_generator(generate_training_from_file(samplesTrain,batch_size),validation_data=generate_test_from_file(samplesTest,batch_size),steps_per_epoch=np.floor(len(samplesTrain)/batch_size), validation_steps=np.floor(len(samplesTest)/batch_size),callbacks=callbacks_list, epochs=50,verbose=1)

I have checked that my generators, generate_training_from_file and generate_test_from_file, are working well.

When very few samples are passed to the network, this one begin to learn very quickly and the training accuracy reaches 1. However, my validation accuaracy is 0.

I don't know if I have any mistake in above line or you think that the problem can be in another place.

Could you help me?

Thank so much.

**Comment by Hs1000** [comment]
Use Image Data Generator go to keras.io and look for flow_from_directory

**Comment by cisi17** [comment]
Hello @Hs1000, 

I am not sure if I can do that because my data are video and a sequence of arrays. I am using a two streams network. 

Do you know, what is the reason, why my code doesn´t work?
And why did you recommend above one?

Thank so much.

**Comment by cisi17** [comment]
Sorry but I only can give you some part.

This is one of the iterator which give you the images, sequences and labels in a batch.

def generate_test_from_file_2(videos,batch_size):
    while True:
        cont=0
        while(cont<len(videos)):
            cont1=cont
            cont2=cont+batch_size
            cont=cont+batch_size
            
            if(cont2>len(videos)):
                cont2=len(videos)
            index=np.arange(cont1,cont2,1)
            for i in range(len(index)):
#                print i
                video=videos[i]
             
                x_frame=getImage(video)
                x_frame=np.expand_dims(x_frame,axis=0)
                x_seq,error=loadArrays(fileSk,30)
                x_seq=np.asarray(x_seq)
                x_seq=np.expand_dims(x_seq,axis=0)
                y=keras.utils.to_categorical(label, num_classes=lenClasses)
                y=np.expand_dims(y,axis=0)
                
                if(i!=0):
                    framesTest=np.append(framesTest,x_frame,axis=0)
                    seqTest=np.append(seqTest,x_seq,axis=0)
                    labelTest=np.append(labelTest,y,axis=0)
                else:
                    framesTest=x_frame
                    seqTest=x_seq
                    labelTest=y
                
#            print (index)
            yield ([framesTest,seqTest],labelTest)


I use a batch size of 8 (beacuse of the memory)
I am surprised because I have 60 classes and when the network only has seen 1000-2000 of 56000 images, training accuracy is 100. And validation accuracy is 0....


I don't know if it is sufficient...

Thanks @Hs1000

**Comment by cisi17** [comment]
Sorry but I only can give you some part.

In the image, thre is one of the iterator which give you the images, sequences and labels in a batch.

![issue](https://user-images.githubusercontent.com/29311113/38366631-b20cc42e-38e0-11e8-900e-df5c8461be24.png)

I use a batch size of 8 (beacuse of the memory)
I am surprised because I have 60 classes and when the network only has seen 1000-2000 of 56000 images, training accuracy is 100. And validation accuracy is 0....

I don't know if it is sufficient...

Thanks @Hs1000
